libHVCAENx527App/src/HVCAENx527.c

```diff
@@ -1345,13 +1399,21 @@ CAENx527SetAllChParVal( HVCRATE *cr, char *pname, void *val)
 			free( pval);
 		}
 #endif	/* CAENHVWrapperVERSION */
-		Busy[cr->crate] = 0;
 		free( chlist);
 		if( retval != CAENHV_OK)
 		{
 			cr->connected = 0;
+#if CAENHVWrapperVERSION / 100 == 2
+			printf("Lost connection to %s@%s: %s (%d)\n", cr->name, cr->IPaddr, CAENHVGetError(cr->name), retval);
+			CAENHVDeinitSystem( cr->name);
+#else
+			printf("Lost connection to %s@%s: %s (%d)\n", cr->name, cr->IPaddr, CAENHVGetError(cr->handle), retval);
+			CAENHV_DeinitSystem( cr->handle);
+#endif	/* CAENHVWrapperVERSION */
```

Comment:
These lines are being repeated in multiple places. Extract this into a function.